ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewManager.java

```diff
@@ -290,6 +334,13 @@ public View getChildAt(ReactViewGroup parent, int index) {
     }
   }
 
+  @Override
+  public void markView(ReactViewGroup parent, int index) {
+    parent.mDeleteMark.remove(index + getPrefixMarkedViewCount(parent, index));
+    parent.mDeleteMark.add(index + getPrefixMarkedViewCount(parent, index),
```

Comment:
The usage of mDeleteMark is quite inefficient here.

Every time you remove an element from a specific index of the ArrayList, you will shift to the left all the elements on the right of the element. Then you will add the element and shift all the same elements but this time to the right. 
See  https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#remove-int- 
and https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#add-int-E-

can you choose a different data-structure?